libs/barista-components/drawer-table/src/drawer-table.ts

```diff
@@ -0,0 +1,188 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AfterContentInit,
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ContentChild,
+  Directive,
+  Input,
+  TemplateRef,
+  ViewChild,
+  ViewContainerRef,
+  ViewEncapsulation,
+} from '@angular/core';
+import {
+  DtHeaderRowDef,
+  DtRowDef,
+  DtTable,
+} from '@dynatrace/barista-components/table';
+
+import { DtDrawer } from '@dynatrace/barista-components/drawer';
+import { isEmpty } from '@dynatrace/barista-components/core';
+
+@Component({
+  selector: 'dt-drawer-table',
+  templateUrl: './drawer-table.html',
+  styleUrls: ['./drawer-table.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.Emulated,
+})
+export class DtDrawerTable<T> implements AfterViewInit, AfterContentInit {
+  /** Columns that stay visible when row is expanded  */
+  @Input() openColumns: string[];
+
+  /** Reference to the drawer element  */
+  @ViewChild(DtDrawer, { static: true }) private drawer: DtDrawer;
+
+  /** Reference to the drawer container  */
+  @ViewChild('container', { read: ViewContainerRef, static: false })
+  private container: ViewContainerRef;
+
+  /** Template reference for the drawer content. */
+  @ContentChild(TemplateRef, { static: false }) drawerContentRef: TemplateRef<{
```

Comment:
Can be marked as internal and prefixed with `_` or even private

Reply:
If you declare your own directive for the templateRef and read the templateRef, it will no longer be necessary for the ngTemplate to be the first child:

pseudo code example: 
```typescript
@ContentChild(DtDrawerContentRef, { static: false, read: TemplateRef }) drawerContentRef: TemplateRef<{ $implicit: T; }>;
```
and in the template: 
```html
...
<ng-template DtDrawerContentRef>...</ng-templae>
...
```

Reply:
I tried this and it still returns undefined, couldn't find the template again.

Reply:
Hi @madina705!

It worked for me when I tested it, here is what I changed and in which files: 

drawer-table.ts added: 
```typescript
@Directive({
  selector: '[dtDrawerContentRef]',
})
export class DtDrawerContentRef {}
```

drawer-table.ts L63-67 changed to:
```typescript
/** @internal Template reference for the drawer content. */
  @ContentChild(DtDrawerContentRef, { static: false, read: TemplateRef })
  private _drawerContentRef: TemplateRef<{
    $implicit: T;
  }>;
```

drawer-table-module.ts L17 changed to:
```typescript
import { DtDrawerRowDef, DtDrawerTable, DtDrawerContentRef } from './drawer-table';
```

drawer-table-module.ts L25-26 changed to:
```typescript
  declarations: [DtDrawerTable, DtDrawerRowDef, DtDrawerContentRef],
  exports: [DtDrawerTable, DtDrawerRowDef, DtDrawerContentRef],
```

drawer-table-demo.component.html, moved ng-template div to the end of the dt-drawer-table and changed to:
```html
<ng-template dtDrawerContentRef let-row>
   <div class="row_col1">{{ row.col1 }} AAA</div>
</ng-template>
```

You may have forgotten to add the new directive to the declarations in the module, which sadly does not throw any error...

Reply:
Hi @tomheller,

Yes it worked, you are right I forgot to add the new directive to the declarations in the module. Thank you very much :)